prov/rxm/src/rxm_init.c

```diff
@@ -199,6 +202,21 @@ static void rxm_alter_info(const struct fi_info *hints, struct fi_info *info)
 			if (hints->mode & FI_BUFFERED_RECV)
 				cur->mode |= FI_BUFFERED_RECV;
 
+			if (hints->caps & FI_ATOMIC) {
+				cur->tx_attr->msg_order &= ~(FI_ORDER_RAR |
+						FI_ORDER_RAW | FI_ORDER_WAR |
+						FI_ORDER_WAW | FI_ORDER_SAR |
+						FI_ORDER_SAW);
+				cur->rx_attr->msg_order &= ~(FI_ORDER_RAR |
+						FI_ORDER_RAW | FI_ORDER_WAR |
+						FI_ORDER_WAW | FI_ORDER_SAR |
+						FI_ORDER_SAW);
```

Comment:
Good point. I've added a routine to verify there are not atomic conflicts in the hints that is part of rxm_getinfo() now. If FI_ATOMIC is requested and MSG Order emulated RxM atomics can not support, it will immediately return -FI_EINVAL. If this is not suitable let me know what is preferable.